pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go

```diff
@@ -57,40 +60,65 @@ type nodeStatusUpdater struct {
 }
 
 func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
+	var nodeIssues []types.NodeName
 	// TODO: investigate right behavior if nodeName is empty
 	// kubernetes/kubernetes/issues/37777
 	nodesToUpdate := nsu.actualStateOfWorld.GetVolumesToReportAttached()
 	for nodeName, attachedVolumes := range nodesToUpdate {
-		nodeObj, err := nsu.nodeLister.Get(string(nodeName))
-		if errors.IsNotFound(err) {
-			// If node does not exist, its status cannot be updated.
-			// Do nothing so that there is no retry until node is created.
-			klog.V(2).Infof(
-				"Could not update node status. Failed to find node %q in NodeInformer cache. Error: '%v'",
-				nodeName,
-				err)
-			continue
-		} else if err != nil {
-			// For all other errors, log error and reset flag statusUpdateNeeded
-			// back to true to indicate this node status needs to be updated again.
-			klog.V(2).Infof("Error retrieving nodes from node lister. Error: %v", err)
-			nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
-			continue
+		err := nsu.processNodeVolumes(nodeName, attachedVolumes)
+		if err != nil {
+			nodeIssues = append(nodeIssues, nodeName)
 		}
+	}
+	if len(nodeIssues) > 0 {
+		return fmt.Errorf("%v could not be updated", nodeIssues)
```

Comment:
We can also return a generic error so as we are not logging 100-300 concatenated node names in logs at once. Something like - `fmt.Errorf("Unable to update %d  nodes for volume attach status")` ? We are anyways logging the error  from individual nodes in `processNodeVolumes` functions.

Reply:
Sure I can update the messages